spec/unit/functions/epp_spec.rb

```diff
@@ -27,9 +27,14 @@
       expect(eval_template_with_args("<%= $phantom == dragos %>", 'phantom' => 'dragos')).to eq("true")
     end
 
-    it "can use values from the enclosing scope for defaults" do
+    it "can use values from the global scope for defaults" do
       scope['phantom'] = 'of the opera'
-      expect(eval_template("<%- |$phantom = $phantom| -%><%= $phantom %>")).to eq("of the opera")
+      expect(eval_template("<%- |$phantom = $::phantom| -%><%= $phantom %>")).to eq("of the opera")
+    end
+
+    it "can use values from the enclosing scope for defaults" do
```

Comment:
Well, it lies, it tests that it can look up using a relative variable using an enclosing scope that also happens to be the global scope. seems a nested scope that is enclosing should be used and that there is a different value in the global scope that it does not select.